src/libs/ViolationsUtils.ts

```diff
@@ -11,10 +11,10 @@ const ViolationsUtils = {
      * violations.
      */
     getViolationsOnyxData(
-        transaction: Transaction,
+        updatedTransaction: Transaction,
         transactionViolations: TransactionViolation[],
         policyRequiresTags: boolean,
-        policyTags: PolicyTags,
+        policyTagList: MaybePolicyTagList,
```

Comment:
@cead22 In testing this, I found a few more issues. `policyTagList` can be an empty object, if there are no tags. I updated types to reflect that.

Similarly, right now we're assuming `transactionViolations` is always going to be defined, but there aren't very solid guarantees of that. We're relying on `policy` being set to mean that `transactionViolations` is also going to exist in onyx on updates and passing that directly through. I tried to see what a non-existent `transactionViolations` would return from `withOnyx` (i.e., is it `undefined` or an empty array or object), but I found that even on non-policy requests, they were getting `transactionViolations` keys with `cashExpenseWithNoReceipt`. Is that expected, or do we need to handle potentially empty `transactionViolations` values?

Reply:
FWIW I would generally recommend making most types nullable in a case where we don't have full type-safety across the project yet, since you can't guarantee what's being passed in, but I'm not sure that's the approach being taken in the TS migration.